libs/SalesforceAnalytics/SalesforceAnalyticsTests/SFSDKLoggerTests.m

```diff
@@ -0,0 +1,186 @@
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <XCTest/XCTest.h>
+#import "SFSDKLogger.h"
+
+static NSString * const kTestComponent1 = @"TestComponent1";
+static NSString * const kTestComponent2 = @"TestComponent2";
+static NSString * const kTestLogLine1 = @"This is test log line 1!";
+static NSString * const kTestLogLine2 = @"This is test log line 2!";
+static NSString * const kTestLogLine3 = @"This is test log line 3!";
+static NSString * const kTestLogLine4 = @"This is test log line 4!";
+unsigned long long const kDefaultMaxFileSize = 1024 * 1024; // 1 MB.
+
+@interface SFSDKLoggerTests : XCTestCase
+
+@end
+
+@implementation SFSDKLoggerTests
+
+- (void)setUp {
+    [super setUp];
+    [SFSDKLogger flushAllComponents];
+    [NSThread sleepForTimeInterval:1.0]; // Flushing the log file is asynchronous.
+    
+}
+
+- (void)tearDown {
+    [SFSDKLogger flushAllComponents];
+    [NSThread sleepForTimeInterval:1.0]; // Flushing the log file is asynchronous.
+    [super tearDown];
+}
+
+/**
+ * Test for setting maximum size of log file.
+ */
+- (void)testSetMaxSize {
+    SFSDKLogger *logger = [SFSDKLogger sharedInstanceWithComponent:kTestComponent1];
+    XCTAssertEqual(kDefaultMaxFileSize, logger.fileLogger.maximumFileSize, @"Max size didn't match expected max size");
+    long long newMaxSize = 1024;
+    logger.fileLogger.maximumFileSize = newMaxSize;
+    XCTAssertEqual(newMaxSize, logger.fileLogger.maximumFileSize, @"Max size didn't match expected max size");
+}
+
+/**
+ * Test for flushing the log file.
+ */
+- (void)testFlushLogFile {
+    SFSDKLogger *logger = [SFSDKLogger sharedInstanceWithComponent:kTestComponent1];
+    XCTAssertEqualObjects(nil, [logger.fileLogger readFile], @"Log file should be empty");
+    [logger.logger log:NO message:[self messageForLogLine:kTestLogLine1]];
```

Comment:
It feels like we should decorate methods like `[DDLog  log:message:]` in the decorator class, if we're not going to go with a model of subclassing `DDLog`.  It seems cumbersome to have to call the underlying logger for what should be first-class operations on the decorating logging facility.